lib/python/src/bailo/helper/model.py

```diff
@@ -263,3 +282,175 @@ def __unpack_mc(self, res):
             self.model_card = res["metadata"]
         except KeyError:
             self.model_card = None
+
+
+class Experiment:
+    """Represent an experiment locally.
+
+    :param model: A Bailo model object which the experiment is being run on
+    :param raw: Raw information about the experiment runs
+    """
+
+    def __init__(
+        self,
+        model: Model,
+    ):
+        self.model = model
+        self.raw = []
+        self.temp_dirs = []
+
+    @classmethod
+    def create(
+        cls,
+        model: Model,
+    ) -> Experiment:
+        """Create an experiment locally.
+
+        :param model: A Bailo model object which the experiment is being run on
+        :return: Experiment object
+        """
+
+        return cls(model=model)
+
+    def start_run(self, is_mlflow: bool | None = False):
+        """Starts a new experiment run.
+
+        :param is_mlflow: Marks a run as MLFlow
+        """
+        try:
+            self.run = self.run + 1
+        except:
+            self.run = 0
+
+        self.run_data = {"run": self.run, "params": [], "metrics": [], "artifacts": [], "dataset": ""}
+
+        self.raw.append(self.run_data)
+
+        if not is_mlflow:
+            print(f"Bailo tracking run {self.run}.")
+
+    def log_params(self, params: dict[str, Any]):
+        """Logs parameters to the current run.
+
+        :param params: Dictionary of parameters to be logged
+        """
+        self.run_data["params"].append(params)
+
+    def log_metrics(self, metrics: dict[str, Any]):
+        """Logs metrics to the current run.
+
+        :param metrics: Dictionary of metrics to be logged
+        """
+        self.run_data["metrics"].append(metrics)
+
+    def log_artifacts(self, artifacts: list):
+        """Logs artifacts to the current run.
+
+        :param artifacts: A list of artifact paths to be logged
+        """
+        self.run_data["artifacts"] = self.run_data["artifacts"] + artifacts
+
+    def log_dataset(self, dataset: str):
+        """Logs a dataset to the current run.
+
+        :param dataset: Arbitrary title of dataset
+        """
+        self.run_data["dataset"] = dataset
+
+    def from_mlflow(self, tracking_uri: str, experiment_id: str):
+        """Imports information from an MLFlow Tracking experiment.
+
+        :param tracking_uri: MLFlow Tracking server URI
+        :param experiment_id: MLFlow Tracking experiment ID
+        :raises ImportError: Import error if MLFlow not installed
+        """
+        if ml_flow == 1:
+            client = mlflow.tracking.MlflowClient(tracking_uri=tracking_uri)
```

Comment:
For readability I'd argue it makes sense to keep it as `import mlfow`. Since there are other components used (e.g. `mlflow.artifacts.download_artifacts`. Likely to use more components in future too